boot/assets.go

```diff
@@ -413,30 +416,35 @@ func (o *TrustedAssetsUpdateObserver) Observe(op gadget.ContentOperation, affect
 	}
 	switch op {
 	case gadget.ContentUpdate:
-		return o.observeUpdate(whichBootloader, isRecovery, root, realSource, relativeTarget)
+		return o.observeUpdate(whichBootloader, isRecovery, root, relativeTarget, data)
 	case gadget.ContentRollback:
-		return o.observeRollback(whichBootloader, isRecovery, root, realSource, relativeTarget)
+		return o.observeRollback(whichBootloader, isRecovery, root, relativeTarget, data)
 	default:
 		// we only care about update and rollback actions
 		return false, nil
 	}
 }
 
-func (o *TrustedAssetsUpdateObserver) observeUpdate(bl bootloader.Bootloader, recovery bool, root, realSource, relativeTarget string) (bool, error) {
+func (o *TrustedAssetsUpdateObserver) observeUpdate(bl bootloader.Bootloader, recovery bool, root, relativeTarget string, data *gadget.ContentChange) (bool, error) {
 	modeenvBefore, err := o.modeenv.Copy()
 	if err != nil {
 		return false, fmt.Errorf("cannot copy modeenv: %v", err)
 	}
 
-	ta, err := o.cache.Add(realSource, bl.Name(), filepath.Base(relativeTarget))
+	ta, err := o.cache.Add(data.After, bl.Name(), filepath.Base(relativeTarget))
```

Comment:
we need to make sure that Before is also tracked if it wasn't already, is this done in a follow up?

Reply:
I left that for a followup. Wanted to keep this PR simple as the patch is mostly a mechanical change.

Reply:
ok, thx